src/ContentRepository/ApplicationModel/ActionFramework.cs

```diff
@@ -175,12 +175,31 @@ public static IEnumerable<ActionBase> GetActions(Content context, string scenari
                 // if the scenario name is given, try to load actions in that scenario
                 var sc = ScenarioManager.GetScenario(scenario, scenarioParameters);
                 if (sc != null)
+                {
                     return sc.GetActions(context, backUri);
+                }
             }
-
             return GetActionsFromContentRepository(context, scenario, backUri);
         }
 
+        public static IEnumerable<ActionBase> GetActions(Content context, string[] scenarios, string backUri)
+        {
+            if (scenarios != null && scenarios.Length>0)
+            {
+                var actions = new List<ActionBase>();
+                foreach (var scenario in scenarios)
+                {
+                    var sc = ScenarioManager.GetScenario(scenario, null);
+                    if (sc != null)
+                    {
+                        actions.AddRange(sc.GetActions(context, backUri));

```

Comment:
A potential bug here: actions may turn up multiple times in the final list if they are present in multiple scenarios.